src/Corax/Querying/MultiTermMatches/IndexSearcher.MultiTermMatches.In.cs

```diff
@@ -84,6 +106,7 @@ private MultiTermMatch InQuery<TTermType>(in FieldMetadata field, List<TTermType
     //In this case, when we get more conditions, we have to quit building the tree and manually check the entries with UnaryMatch.
     private IQueryMatch AllInQuery<TTerm>(in FieldMetadata field, HashSet<TTerm> allInTerms, bool skipEmptyItems = false, in CancellationToken cancellationToken = default)
     {
+        Debug.Assert(typeof(TTerm) == typeof(InParameter) || typeof(TTerm) == typeof(Slice) || typeof(TTerm) == typeof(string));
         const int maximumTermMatchesHandledAsTermMatches = 4;
```

Comment:
Should throw with `UnsupportedException` if it is not in the list of valid cases (check will be evicted anyways by the JIT).